compiler/rustc_middle/src/query/plumbing.rs

```diff
@@ -24,8 +23,8 @@ pub struct DynamicQuery<'tcx, C: QueryCache> {
     pub eval_always: bool,
     pub dep_kind: DepKind,
     pub handle_cycle_error: HandleCycleError,
-    pub query_state: FieldOffset<QueryStates<'tcx>, QueryState<C::Key>>,
-    pub query_cache: FieldOffset<QueryCaches<'tcx>, C>,
+    pub query_state: isize,
```

Comment:
this should store `usize` and use byte_add instead

Reply:
Thanks, missed that those exist, they are indeed the better option. The change is on the way.